File: app/src/main/java/beg/hr/rxredux/java/redux/presentation/ViewDriverComponent.java
package beg.hr.rxredux.java.redux.presentation;

import android.view.View;

/** Created by juraj on 17/03/2017. */
public abstract class ViewDriverComponent<VD extends ViewDriver> {

  private final VD driver;

  public ViewDriverComponent(VD driver) {
    this.driver = driver;
  }

  protected VD driver() {
    return driver;
  }

  public View view() {
    return driver.view();
  }
}

File: app/src/main/java/beg/hr/rxredux/kotlin/util/ui/ViewDriverComponent.kt
package beg.hr.rxredux.kotlin.util.ui

import android.view.View

/**
 * Created by juraj on 26/03/2017.
 */
abstract class ViewDriverComponent<in S, out V : View, C, out VD : ViewDriver<V, S, C>>(val driver: VD) : ViewComponent<V> {
    override fun view(): V = driver.view
}
